src/portal_visualization/builders/anndata_builders.py

```diff
@@ -74,17 +76,35 @@ def get_conf_cells(self):
             gene_alias=gene_alias
         ))
 
-        vc = self._setup_anndata_view_config(vc, dataset)
+        vc = self._setup_anndata_view_config(vc, dataset, marker_gene)
         return get_conf_cells(vc)
 
-    def _setup_anndata_view_config(self, vc, dataset):
-        vc.add_view(cm.SCATTERPLOT, dataset=dataset, mapping="UMAP", x=0, y=0, w=4, h=6)
-        vc.add_view(cm.CELL_SET_EXPRESSION, dataset=dataset, x=4, y=0, w=5, h=6)
-        vc.add_view(cm.CELL_SETS, dataset=dataset, x=9, y=0, w=3, h=3)
-        vc.add_view(cm.GENES, dataset=dataset, x=9, y=4, w=3, h=3)
-        vc.add_view(cm.HEATMAP, dataset=dataset, x=0, y=6, w=12, h=4)
+    def _setup_anndata_view_config(self, vc, dataset, marker_gene=None):
+        scatterplot = vc.add_view(
+            cm.SCATTERPLOT, dataset=dataset, mapping="UMAP", x=0, y=0, w=self._scatterplot_w, h=6)
+        cell_sets = vc.add_view(
+            cm.CELL_SETS, dataset=dataset, x=9, y=0, w=3, h=3)
+        gene_list = vc.add_view(
+            cm.GENES, dataset=dataset, x=9, y=4, w=3, h=3)
+        cell_sets_expr = vc.add_view(
+            cm.CELL_SET_EXPRESSION, dataset=dataset, x=7, y=6, w=5, h=4)
```

Comment:
I saw https://github.com/vitessce/vitessce-python/issues/197 but to prevent the components from changing their ordering in the config fixtures, you could also run the `vc.add_view` lines in the original ordering of SCATTERPLOT, CELL_SET_EXPRESSION, CELL_SETS, GENES, HEATMAP.

Reply:
Thanks: I don't think I can do that here because there were actually two different creation orders: The class and the subclass ran `add_view` in different orders... and for readability, I also wanted to move the spatial `add_view` to the end.